cpp/src/io/parquet/compact_protocol_writer.cpp

```diff
@@ -84,31 +82,28 @@ size_t CompactProtocolWriter::write(IntType const& integer)
 size_t CompactProtocolWriter::write(LogicalType const& logical_type)
 {
   CompactProtocolFieldWriter c(*this);
-  auto const isset = logical_type.isset;
-  if (isset.STRING) {
-    c.field_struct(1, logical_type.STRING);
-  } else if (isset.MAP) {
-    c.field_struct(2, logical_type.MAP);
-  } else if (isset.LIST) {
-    c.field_struct(3, logical_type.LIST);
-  } else if (isset.ENUM) {
-    c.field_struct(4, logical_type.ENUM);
-  } else if (isset.DECIMAL) {
-    c.field_struct(5, logical_type.DECIMAL);
-  } else if (isset.DATE) {
-    c.field_struct(6, logical_type.DATE);
-  } else if (isset.TIME) {
-    c.field_struct(7, logical_type.TIME);
-  } else if (isset.TIMESTAMP) {
-    c.field_struct(8, logical_type.TIMESTAMP);
-  } else if (isset.INTEGER) {
-    c.field_struct(10, logical_type.INTEGER);
-  } else if (isset.UNKNOWN) {
-    c.field_struct(11, logical_type.UNKNOWN);
-  } else if (isset.JSON) {
-    c.field_struct(12, logical_type.JSON);
-  } else if (isset.BSON) {
-    c.field_struct(13, logical_type.BSON);
+  switch (logical_type.type) {
+    case LogicalType::STRING: c.field_empty_struct(LogicalType::STRING); break;
+    case LogicalType::MAP: c.field_empty_struct(LogicalType::MAP); break;
+    case LogicalType::LIST: c.field_empty_struct(LogicalType::LIST); break;
+    case LogicalType::ENUM: c.field_empty_struct(LogicalType::ENUM); break;
+    case LogicalType::DECIMAL:
+      c.field_struct(LogicalType::DECIMAL, logical_type.decimal_type.value());
+      break;
+    case LogicalType::DATE: c.field_empty_struct(LogicalType::DATE); break;
+    case LogicalType::TIME:
+      c.field_struct(LogicalType::TIME, logical_type.time_type.value());
+      break;
+    case LogicalType::TIMESTAMP:
+      c.field_struct(LogicalType::TIMESTAMP, logical_type.timestamp_type.value());
+      break;
+    case LogicalType::INTEGER:
+      c.field_struct(LogicalType::INTEGER, logical_type.int_type.value());
+      break;
+    case LogicalType::UNKNOWN: c.field_empty_struct(LogicalType::UNKNOWN); break;
+    case LogicalType::JSON: c.field_empty_struct(LogicalType::JSON); break;
+    case LogicalType::BSON: c.field_empty_struct(LogicalType::BSON); break;
```

Comment:
Most of these could be `c.field_empty_struct(logical_type.type);` ?
I wonder if we should simply catch all of these under `default`.

Reply:
I'd prefer to enumerate them and fall through, and leave defaut as a no-op in case there's a weird value set for type.  Probably more of a concern on the read side where new logical types we don't understand can get added.

Reply:
Sounds good for the most part - should the default throw instead?

Reply:
Hmm, on the write side...yeah, that's probably a good idea.  We likely want to know when the struct isn't properly initialized.